### PR TITLE
Add a condition to retrieve correct image URI for xgboost

### PR DESCRIPTION
**Pull request by nskool**
*Issue #, if available:*
#1932 

*Description of changes:*
Rename the .json that holds registry information for neo containers.

*Testing done:*
```
tox -e flake8,pylint,twine,black-check --parallel all
tox -e py38 tests/unit
```

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/model.py [hunk 1]
```diff
@@ -274,9 +274,17 @@ def _compilation_image_uri(self, region, target_instance_type, framework, framew
...
+        
+        if framework == "xgboost":
+            framework_suffix = "-neo"
+        else:
+            framework_prefix = "inferentia-" if target_instance_type.startswith("ml_inf") else "neo-"
+
+
         return image_uris.retrieve(
-            "{}{}".format(framework_prefix, framework),
+            "{}{}".format(framework_prefix, framework, framework_suffix),
```
Comment (icywang86rui) [inline src/sagemaker/model.py:287]: I don't think this does what you want it to do

```
>>> "{}{}".format("",'2','3')
'2'
```

File: src/sagemaker/model.py [hunk 2]
```diff
@@ -274,9 +274,17 @@ def _compilation_image_uri(self, region, target_instance_type, framework, framew
             framework (str): The framework name.
             framework_version (str): The framework version.
         """
-        framework_prefix = "inferentia-" if target_instance_type.startswith("ml_inf") else "neo-"
+        framework_prefix = ""
+        framework_suffix = ""
+        
+        if framework == "xgboost":
```
Comment (icywang86rui) [inline src/sagemaker/model.py:280]: i think we need this to include the image_classifiction config as well

**Review by icywang86rui — CHANGES_REQUESTED** [review]

File: src/sagemaker/model.py (hunk 1 above)
Comment (nskool) [reply src/sagemaker/model.py]: Addressed.

File: src/sagemaker/model.py (hunk 2 above)
Comment (icywang86rui) [reply src/sagemaker/model.py:280]: image-classifiction is not in the allowed framework for neo list. we probably have a bug here. This should be fixed.

File: src/sagemaker/model.py [hunk 3]
```diff
@@ -274,9 +274,19 @@ def _compilation_image_uri(self, region, target_instance_type, framework, framew
             framework (str): The framework name.
             framework_version (str): The framework version.
         """
-        framework_prefix = "inferentia-" if target_instance_type.startswith("ml_inf") else "neo-"
+        framework_prefix = ""
+        framework_suffix = ""
+
+        if framework == "xgboost":
+            framework_suffix = "-neo"
+        else:
```
Comment (icywang86rui) [inline src/sagemaker/model.py:282]:
```suggestion
        elif target_instance_typ.startswith("ml_inf"):
            framework_prefix = "inferentia-"
        else:
            framework_prefix = "neo-"
```

**Review by icywang86rui — APPROVED** [review]

**Pull request merged**